packages/panzer/disc-fe/src/evaluators/Panzer_Integrator_CurlBasisDotVector.hpp

```diff
@@ -40,92 +40,339 @@
+        const panzer::IntegrationDescriptor& id,
+        const int&                           spaceDim    = 3,
+        const double&                        multiplier  = 1,
+        const std::vector<PHX::FieldTag>&    multipliers =
+          std::vector<PHX::FieldTag>());
+
+      /**
+       *  \brief Post-Registration Setup.
+       *
+       *  Sets the `Kokkos::View`s for all of the field multipliers, sets the
+       *  basis index, and sets up the field that will be used to build up the
+       *  result of the integration.
+       *
+       *  \param[in] sd Essentially a list of `Workset`s, which are collections
+       *                of cells (elements) that all live on a single process.
+       *  \param[in] fm The `FieldManager` used to create the field to build up
+       *                the result.
+       */
+      void
+      postRegistrationSetup(
+        typename Traits::SetupData sd,
+        PHX::FieldManager<Traits>& fm);
+
+      /**
+       *  \brief Evaluate Fields.
+       *
+       *  This actually performs the integration using a handful of functors in
+       *  `Kokkos::parallel_for`s, looping over the cells in the `Workset`.
+       *
+       *  \param[in] workset The `Workset` on which you're going to do the
+       *                     integration.
+       */
+      void
+      evaluateFields(
+        typename Traits::EvalData d);
+
+    private:
+
+      /**
+       *  \brief Get Valid Parameters.
+       *
+       *  Get all the parameters that we support such that the `ParameterList`
+       *  Constructor can do some validation of the input `ParameterList`.
+       *
+       *  \returns A `ParameterList` with all the valid parameters (keys) in
+       *           it.  The values tied to those keys are meaningless default
+       *           values.
+       */
+      Teuchos::RCP<Teuchos::ParameterList> getValidParameters() const;
+
+      /**
+       *  \brief The scalar type.
+       */
+      using ScalarT = typename EvalT::ScalarT;
+
+      /**
+       *  \brief An `enum` determining the behavior of this `Evaluator`.
+       *
+       *  This `Evaluator` will compute the result of its integration and then:
+       *  - CONTRIBUTES:  contribute it to a specified residual, not saving
+       *                  anything; or
+       *  - EVALUATES:    save it under a specified name for future use.
+       */
+      const panzer::EvaluatorStyle evalStyle_;
 
-    Integrator_CurlBasisDotVector(
-      const Teuchos::ParameterList& p);
+      /**
+       *  \brief A flag indicating whether or not to use the descriptor
+       *         interface.
+       */
+      bool useDescriptors_;
 
-    void
-    postRegistrationSetup(
-      typename Traits::SetupData d,
-      PHX::FieldManager<Traits>& fm);
+      /**
+       *  \brief The `BasisDescriptor` for the basis to use.
+       */
+      panzer::BasisDescriptor bd_;
 
-    void
-    evaluateFields(
-      typename Traits::EvalData d);
+			/**
```

Comment:
Looks like the indenting on a bunch of these comments is messed up and pushed too far to right. Same for next 8 functions.

Reply:
Blast!  How did tabs wind up in here?  I'll need to give my `.vimrc` a stern talking-to.